askametric/query_processor/query_processor.py

```diff
@@ -6,55 +6,83 @@
 from .query_processing_prompts import (
     create_best_columns_prompt,
     create_best_tables_prompt,
+    create_conversation_summary_prompt,
     create_final_answer_prompt,
     create_sql_generating_prompt,
+    english_translation_prompt,
     get_query_language_prompt,
 )
 from .tools import SQLTools, get_tools
+from .schemas import Prompts, UserQueryResponse
 
 
 class LLMQueryProcessor:
     """Processes the user query and returns the final answer."""
 
     def __init__(
         self,
-        query: dict,
         asession: AsyncSession,
-        which_db: str,
+        metric_db_id: str,
+        db_type: str,
         llm: str,
         guardrails_llm: str,
         sys_message: str,
         db_description: str,
         column_description: str,
+        indicator_vars: list,
         num_common_values: int,
+        context_length: int,
     ) -> None:
-        """Initialize the QueryProcessor class."""
-        self.query = query
+        """
+        Initialize the QueryProcessor class.
+
+        Args:
+            asession: The SQLAlchemy AsyncSession object.
+            metric_db_id: The database id to query.
+            llm: The LLM model to use.
+            guardrails_llm: The guardrails LLM model to use.
+            sys_message: The system message to use.
+            db_description: The description of the database.
+            column_description: The description of the columns.
+            indicator_vars: The indicator variables.
+            num_common_values: The number of common values to get.
+            context_length: The number of previous queries to consider.
+        """
         self.asession = asession
-        self.which_db = which_db
+        self.metric_db_id = metric_db_id
+        self.db_type = db_type
         self.tools: SQLTools = get_tools()
-        self.temperature = 0.1
+        self.temperature: float = 0.1
         self.llm = llm
         self.system_message = sys_message
         self.table_description = db_description
         self.column_description = column_description
+        self.indicator_vars = indicator_vars
         self.num_common_values = num_common_values
-        self.guardrails: LLMGuardRails = LLMGuardRails(
-            guardrails_llm, self.system_message
-        )
-        self.cost = 0.0
-        self.language_prompt = ""
-        self.query_language = ""
-        self.script = ""
-        self.best_tables: str = ""
-        self.best_columns: str = ""
-        self.sql_query: str = ""
-        self.final_answer: str = ""
-        self.relevant_schemas: str = ""
-        self.best_tables_prompt: str = ""
-        self.best_columns_prompt: str = ""
-        self.sql_generating_prompt: str = ""
-        self.final_answer_prompt: str = ""
+        self.context_length = context_length
+
+        self.context: list[dict] = []
+        self.context_keys = ["text_response", "sql_query", "relevant_schemas"]
```

Comment:
I wonder whether it is useful to add table schema as past relevant query? Does it not do well if not included? Maybe helpful in answering questions which were earlier answered incorrectly?

Maybe to tune multi-turn we can again go the agentic route and see what's the best way?

Of course for future tickets.

Reply:
removed the relevant schemas for now -- the questions on the demo are answered properly even without it.
will use the agentic flow to test this more thoroughly -- future ticket